src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProject.cs

```diff
@@ -333,33 +336,66 @@ private IList<string> GetConfigFilePaths(ISettings settings)
             return settings.GetConfigFilePaths();
         }
 
-        private static PackageReference[] GetPackageReferences(PackageSpec packageSpec)
+        private async Task<IEnumerable<PackageReference>> GetPackageReferences(PackageSpec packageSpec)
         {
             var frameworkSorter = new NuGetFrameworkSorter();
 
+            var lockFilePath = await GetAssetsFilePathAsync();
+            var fileInfo = new FileInfo(lockFilePath);
+            PackageSpec assetsPackageSpec = default;
+            IList<LockFileTarget> targets = default;
+
+            if (fileInfo.Exists && fileInfo.LastWriteTimeUtc > _lastTimeAssetsModified)
```

Comment:
I don't think you are caching anything with this approach.

Consider the happy scenario. 

* 1st pass. We have not read the assets file before. We now read it...so we return the correct version.
* 2nd pass. We have read the assets file before, we now read it, we end up returning the minimum version.